### PR TITLE
feat(user): add route to get user details

### LINKED ISSUE #4509: feat(user): get user information (home api)
Create new api to get user details.

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Add support to get user details!

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
closes #4509 

## How did you test it?
Use the curl to get user info:
```
curl --location 'http://localhost:8080/user' \
--header 'Authorization: Bearer JWT' \
--data ''
```

Response will be, something like:
```
{
    "merchant_id": "merchant_1714396883",
    "name": "test1",
    "email": "apoorvdixit88+555@gmail.com",
    "verification_days_left": null,
    "role_id": "merchant_view_only",
    "org_id": "org_HomVKdwu2HGTi6jUvvnz"
}
```
## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router_env/src/logger/types.rs [hunk 1]
```diff
@@ -350,6 +350,8 @@ pub enum Flow {
     UserMerchantAccountList,
     /// Get details of a user in a merchant account
```
Comment (ThisIsMani) [inline crates/router_env/src/logger/types.rs:351]: I think this comment should be changed.

File: crates/router/src/core/user.rs [hunk 2]
```diff
@@ -71,6 +71,38 @@ pub async fn signup_with_merchant_id(
...
+        .store
+        .find_user_by_id(&user_from_token.user_id)
+        .await
+        .map_err(|e| {
+            if e.current_context().is_db_not_found() {
+                e.change_context(UserErrors::InvalidCredentials)
+            } else {
+                e.change_context(UserErrors::InternalServerError)
+            }
+        })?
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:88]: This should be `InternalServerError` always as we are getting the `user_id` from the token.

File: crates/router/src/core/user.rs [hunk 3]
```diff
@@ -71,6 +71,38 @@ pub async fn signup_with_merchant_id(
...
+        .into();
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:89]: `user_from_token` has a function called `get_user_from_db()` which does the same thing. You can replace this whole code with that.

File: crates/api_models/src/user.rs [hunk 4]
```diff
@@ -149,13 +149,25 @@ pub struct UserDetails {
...
 }
 
+#[derive(serde::Serialize, Debug, Clone)]
+pub struct GetUserDetailsResponse {
+    pub merchant_id: String,
+    pub name: Secret<String>,
+    pub email: pii::Email,
+    pub verification_days_left: Option<i64>,
+    pub role_id: String,
+    //this field is added for audit/debug reasons
```
Comment (ThisIsMani) [inline crates/api_models/src/user.rs:159]: I know that we use this comment every where but from now let's correct it.
```suggestion
    // This field is added for audit/debug reasons
```

File: crates/router/src/core/user.rs (hunk 2 above)
Comment (ThisIsMani) [reply crates/router/src/core/user.rs:88]: See next comment before changing this.

**Review by ThisIsMani — APPROVED** [review]

**Review by racnan — APPROVED** [review]

**Review by ShivanshMathurJuspay — APPROVED** [review]

**Pull request merged**